src/js/pebble-js-app.js

```diff
@@ -27,8 +31,24 @@ Pebble.addEventListener("appmessage", function(e) {
 });
 
 function downloadBinaryResource(imageURL, callback, errorCallback) {
+
+
+    /*POSTing the image URL to the server for conversion.
+      This conversion is only for Pebble Time. On an original
+      Pebble the dithering should convert the image to have 
+      to 2 colors*/
+    var req1 = new XMLHttpRequest();
+    req1.open("POST",toServer, false);
+    imageURL = "imageurl=" + imageURL+"&PebbleType=0";
+    req1.setRequestHeader("Content-type", "imageURL");
+    req1.setRequestHeader("Content-length", imageURL.length);
+    req1.setRequestHeader("Connection", "close");    
+    req1.send(imageURL);
+
+    //getting converted image, which is always at <server>/Pebble_image.png
     var req = new XMLHttpRequest();
-    req.open("GET", imageURL,true);
+    var img_return = toServer+ "/Pebble_image.png"; 
```

Comment:
What do you think will happen if two clients try to convert an image at roughly the same time?

This is very broken. You need to pass the URL of the image you want to the GET request AND return the converted image directly.